lang/typescript/src/extended-address/splitAddress1WithFallback.ts

```diff
@@ -0,0 +1,35 @@
+import type {Address} from '../types/address';
+import {splitAddressFieldWithFallback} from '../utils/address-fields';
+import {
+  getRegionConfig,
+  getAddress1Regex,
+  getConcatenationRules,
+} from '../utils/regions';
+
+/**
+ * Split an address1 string based on the region specified by
+ * country code, fall back on an address1 regex pattern
+ * @param countryCode 2-letter country code string
+ * @param concatenatedAddress Combined address1 string
+ * @returns Partial Address object containing split address fields or null if
+ * the region does not define an extended address format
+ */
+export function splitAddress1WithFallback(
+  countryCode: string,
+  concatenatedAddress: string,
+): Partial<Address> | null {
```

Comment:
@gabypancu I actually think I prefer just changing the default behavior or making it an optional param to turn on/off the regex fallback. Not a super strong opinion so if you have a good reason this is fine, but the library is new enough and only has one main consumer so I feel like making the decision now to change it to add this fallback always if the user wants to opt-out. At the end of the day all of the split logic is defined in the region yamls so I think we can safely change this and just bump the major version to signify the clear change in scope.

I don't hate just doing `splitAddress1('NL', 'Main street 123', false)` even if its a little clunky if we want to offer a way to turn off the fallback regex.

If we go with a second function we should document this in the README too.

Reply:
This was initially my preference as well. @rochlefebvre brought up a good point that this "fallback" logic is not ideal, and we don't want to draw attention to it. That was the main motivator for having it in a separate function.

Reply:
Yeah just run the pnpm changeset is all. I was recommending to update the README to document usage of the new function but if we want it to be undocumented on purpose then no need. After it gets merged to main a PR will open to handle the release.

Reply:
I think my main concern right now is that this is still a public facing function but it's not really clear why it's there when reading through it without context. So if someone goes to import the documented `splitAddress1` function they will see a semi-documented second `splitAddress1WithFallback` function with a similar name in autocomplete/IntelliSense. I think this might have the opposite effect we want: "Oh I should use the one with the fallback, because it has fallback functionality which is a good thing". We could double-down on the "don't use this" or "it's experimental" and just popping a warning in the docstring comment for the new fallback function.

Imho though I think adding it as an explicit set of options on the existing function by doing `splitAddress1('NL', 'Main 123', { experimentalRegexFallback: true })`. That way it's opt-in and very clear it's potentially not something someone should use without knowing what they're doing. Then if don't want to document it in the README or later we decide we want to no longer expose this option on the public surface of the API we can just remove it.

Reply:
Discussed off-Github - we've aligned on moving the fallback logic into the existing `splitAddress1` function behind an optional `experimentalRegexFallback` flag for the reasons outlined above. We will also release this as a beta feature, given that support for countries will be added incrementally